Added .NET SDK for Microsoft.HybridContainerService

HariPrasadV opened the pull request:
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

HariPrasadV committed:
Added .NET SDK for Microsoft.HybridContainerService

ArthurMa1978 commented on sdk/hybridaks/ci.yaml:
```diff
@@ -0,0 +1,32 @@
+# NOTE: Please refer to https://aka.ms/azsdk/engsys/ci-yaml before editing this file.
+
+trigger:
```
remove this file

HariPrasadV replied:
Done!

ArthurMa1978 commented on sdk/hybridaks/Azure.ResourceManager.HybridContainerService/src/Properties/AssemblyInfo.cs:
```diff
@@ -0,0 +1,9 @@
+
+using System.Runtime.CompilerServices;
+
+[assembly: Azure.Core.AzureResourceProviderNamespace("HybridContainerService")]
```
Microsoft.HybridContainerService

HariPrasadV replied:
Corrected it now

HariPrasadV committed:
Corrected the package name and ci yaml file

fengzhou-msft commented:
Please generate the API file with the command `.\azure-sdk-for-net\eng\scripts\Export-API.ps1 hybridaks`

fengzhou-msft commented on sdk/hybridaks/Azure.ResourceManager.HybridContainerService/src/Generated/AgentPoolResource.cs:
```diff
@@ -0,0 +1,504 @@
+    /// from an instance of <see cref="ArmClient" /> using the GetAgentPoolResource method.
+    /// Otherwise you can get one from its parent resource <see cref="ProvisionedClustersResponseResource" /> using the GetAgentPool method.
+    /// </summary>
+    public partial class AgentPoolResource : ArmResource
```
`AgentPool` is a common resource name. Please add a prefix to make it more descriptive. Either use the `prepend-rp-prefix` if `HybridContainerServiceAgentPool` is a good name to use or use `rename-mapping` to use another name in  `autorest.md`. You can check rename configuration examples in [ContainerService autorest.md](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/containerservice/Azure.ResourceManager.ContainerService/src/autorest.md)

HariPrasadV replied:
Thanks for the suggestion, I've renamed it now

fengzhou-msft commented on sdk/hybridaks/Azure.ResourceManager.HybridContainerService/src/Generated/VirtualNetworkResource.cs:
```diff
@@ -0,0 +1,514 @@
+    /// from an instance of <see cref="ArmClient" /> using the GetVirtualNetworkResource method.
+    /// Otherwise you can get one from its parent resource <see cref="ResourceGroupResource" /> using the GetVirtualNetwork method.
+    /// </summary>
+    public partial class VirtualNetworkResource : ArmResource
```
`VirtualNetwork` is too generic. Please add a prefix to the resource name to make it more descriptive.

HariPrasadV replied:
Thanks for the suggestion, I've renamed it now

fengzhou-msft commented on sdk/hybridaks/Azure.ResourceManager.HybridContainerService/src/Generated/ProvisionedClustersResponseResource.cs:
```diff
@@ -0,0 +1,588 @@
+    /// from an instance of <see cref="ArmClient" /> using the GetProvisionedClustersResponseResource method.
+    /// Otherwise you can get one from its parent resource <see cref="ResourceGroupResource" /> using the GetProvisionedClustersResponse method.
+    /// </summary>
+    public partial class ProvisionedClustersResponseResource : ArmResource
```
The generator takes the response model in swagger as the resource name, but we should get rid of the `Response` suffix as a resource anme. Rename `ProvisionedClustersResponse` to `ProvisionedCluster`. If there is an existing `ProvisionedCluster(s)`, rename that model to something else.

HariPrasadV replied:
Thanks for the suggestion, I've renamed it now

ArthurMa1978 commented:
@HariPrasadV can you update the pr for the comments?

HariPrasadV committed:
Added vnet creation test

HariPrasadV committed:
Added test for provisioned cluster

HariPrasadV committed:
added recorded tests

HariPrasadV committed:
Added Ignore for cluster creation test as it takes more than 15 seconds

HariPrasadV committed:
Addressing PR comments

HariPrasadV commented:
> @HariPrasadV can you update the pr for the comments?

Sorry for the delay, I've addressed all the comments now, and also tested the SDK and it looks good from our side. Please take a look and let me know if there is anything else. Thanks!

HariPrasadV committed:
Added API file

HariPrasadV committed:
Corrected the custom location in test

ArthurMa1978 commented:
Merge this pr first, kick off release once the namespace got [approved](https://github.com/Azure/azure-sdk/issues/5057).
